### PR TITLE
Add an integration test to validate pick mode crash

### PR DESCRIPTION
**Pull request by hultonha**
This change reproduces the situation that caused the pick mode crash (issue #2199) and is related to the fix from https://github.com/o3de/o3de/pull/2884.

With my changes reverted I was able to repro the crash (see call stack below) and with my fix the test runs to completion and passes

```
>	AzToolsFramework.Tests.dll!AzToolsFramework::EditorInteractionSystemComponent::InternalHandleMouseViewportInteraction(const AzToolsFramework::ViewportInteraction::MouseInteractionEvent & mouseInteraction) Line 35	C++
 	[External Code]	
 	AzToolsFramework.Tests.dll!UnitTest::FocusInteractionWidget::event(QEvent * event) Line 90	C++
 	[External Code]	
 	[Inline Frame] AzToolsFramework.Tests.dll!QWidget::setFocus() Line 420	C++
 	[Inline Frame] AzToolsFramework.Tests.dll!AzToolsFramework::ViewportUi::Internal::ViewportUiDisplay::SetUiOverlayContentsAnchored(QPointer<QWidget>) Line 395	C++
 	AzToolsFramework.Tests.dll!AzToolsFramework::ViewportUi::Internal::ViewportUiDisplay::PositionViewportUiElementAnchored(AzToolsFramework::IdType<AzToolsFramework::ViewportUi::ViewportUiIdType> elementId, const QFlags<enum Qt::AlignmentFlag> alignment) Line 333	C++
 	AzToolsFramework.Tests.dll!AzToolsFramework::ViewportUi::Internal::ViewportUiDisplay::AddCluster(AZStd::shared_ptr<AzToolsFramework::ViewportUi::Internal::ButtonGroup> buttonGroup, AzToolsFramework::ViewportUi::Alignment alignment) Line 84	C++
 	AzToolsFramework.Tests.dll!AzToolsFramework::ViewportUi::ViewportUiManager::CreateCluster(AzToolsFramework::ViewportUi::Alignment align) Line 32	C++
 	[External Code]	
 	AzToolsFramework.Tests.dll!AzToolsFramework::EditorTransformComponentSelection::CreateTransformModeSelectionCluster() Line 2487	C++
 	AzToolsFramework.Tests.dll!AzToolsFramework::EditorTransformComponentSelection::EditorTransformComponentSelection(const AzToolsFramework::EditorVisibleEntityDataCache * entityDataCache) Line 1046	C++
 	[Inline Frame] AzToolsFramework.Tests.dll!AZStd::make_unique(const AzToolsFramework::EditorVisibleEntityDataCache * &) Line 44	C++
 	AzToolsFramework.Tests.dll!AzToolsFramework::EditorDefaultSelection::EditorDefaultSelection(const AzToolsFramework::EditorVisibleEntityDataCache * entityDataCache) Line 30	C++
 	[External Code]	
 	[Inline Frame] AzToolsFramework.Tests.dll!AZStd::function_intermediate<std::unique_ptr<AzToolsFramework::ViewportInteraction::InternalViewportSelectionRequests,std::default_delete<AzToolsFramework::ViewportInteraction::InternalViewportSelectionRequests>>,AzToolsFramework::EditorVisibleEntityDataCache const *>::operator()(const AzToolsFramework::EditorVisibleEntityDataCache * &&) Line 607	C++
 	[Inline Frame] AzToolsFramework.Tests.dll!AZStd::function<std::unique_ptr<AzToolsFramework::ViewportInteraction::InternalViewportSelectionRequests,std::default_delete<AzToolsFramework::ViewportInteraction::InternalViewportSelectionRequests>> __cdecl(AzToolsFramework::EditorVisibleEntityDataCache const *)>::operator()(const AzToolsFramework::EditorVisibleEntityDataCache * <args_0>) Line 687	C++
 	AzToolsFramework.Tests.dll!AzToolsFramework::EditorInteractionSystemComponent::SetHandler(const AZStd::function<std::unique_ptr<AzToolsFramework::ViewportInteraction::InternalViewportSelectionRequests,std::default_delete<AzToolsFramework::ViewportInteraction::InternalViewportSelectionRequests>> __cdecl(AzToolsFramework::EditorVisibleEntityDataCache const *)> & interactionRequestsBuilder) Line 57	C++
 	AzToolsFramework.Tests.dll!AzToolsFramework::EditorInteractionSystemComponent::SetDefaultHandler() Line 62	C++
 	[External Code]	
 	AzToolsFramework.Tests.dll!UnitTest::FocusInteractionWidget::event(QEvent * event) Line 79	C++
 	[External Code]	
 	AzToolsFramework.Tests.dll!QTest::mouseEvent(QTest::MouseAction action, QWidget * widget, Qt::MouseButton button, QFlags<enum Qt::KeyboardModifier> stateKey, QPoint pos, int delay) Line 242	C++
 	AzToolsFramework.Tests.dll!QTest::mouseEvent(QTest::MouseAction action, QWidget * widget, Qt::MouseButton button, QFlags<enum Qt::KeyboardModifier> stateKey, QPoint pos, int delay) Line 206	C++
 	[Inline Frame] AzToolsFramework.Tests.dll!QTest::mouseClick(QWidget *) Line 262	C++
 	AzToolsFramework.Tests.dll!UnitTest::EditorTransformComponentSelectionFixture_Focus_is_not_changed_while_switching_viewport_interaction_request_instance_Test::TestBody() Line 223	C++
 	[External Code]	
 	[Inline Frame] AzToolsFramework.Tests.dll!RUN_ALL_TESTS() Line 2341	C++

```

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

File: Code/Framework/AzToolsFramework/Tests/EditorTransformComponentSelectionTests.cpp [hunk 1]
```diff
@@ -188,6 +190,44 @@ namespace UnitTest
...
+                return AZStd::make_unique<AzToolsFramework::EditorPickEntitySelection>(entityDataCache);
+            });
+
+        // When
+        // A mouse event is sent to the focus widget (set to be the render overlay in the viewport ui system)
+        QTest::mouseClick(focusWidget.get(), Qt::MouseButton::LeftButton);
+
+        // Then
+        // focus should not change
+        EXPECT_FALSE(focusWidget->hasFocus());
```
Comment (NicholasVanSickle) [inline Code/Framework/AzToolsFramework/Tests/EditorTransformComponentSelectionTests.cpp:223]: suggestion: We could cache and check that `QApplication::focusWidget` doesn't change to ensure that no other esoteric focus change behavior occurred and we're still on the correct widget

**Review by NicholasVanSickle — APPROVED** [review]
Looks good to me!

**Review by amznestebanpapp — APPROVED** [review]

File: Code/Framework/AzToolsFramework/Tests/EditorTransformComponentSelectionTests.cpp (hunk 1 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/Tests/EditorTransformComponentSelectionTests.cpp:223]: Sure good shout I'll give that a try 👍

**Comment by hultonha** [comment]
Test run

```
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from EditorTransformComponentSelectionFixture
[ RUN      ] EditorTransformComponentSelectionFixture.Focus_is_not_changed_while_switching_viewport_interaction_request_instance
[       OK ] EditorTransformComponentSelectionFixture.Focus_is_not_changed_while_switching_viewport_interaction_request_instance (207 ms)
[----------] 1 test from EditorTransformComponentSelectionFixture (209 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (213 ms total)
[  PASSED  ] 1 test.
```

**Pull request merged**